src/components/structures/login/Login.js

```diff
@@ -232,11 +232,24 @@ module.exports = React.createClass({
         }).done();
     },
 
-    onUsernameChanged: function(username, endOfInput) {
+    onUsernameChanged: function(username) {
         this.setState({ username: username });
-        if (username[0] === "@" && endOfInput) {
+    },
+
+    onUsernameBlur: function(username) {
+        this.setState({ username: username });
+        if (username[0] === "@") {
             const serverName = username.split(':').slice(1).join(':');
-            this._tryWellKnownDiscovery(serverName);
+            try {
+                // we have to append 'https://' to make the URL constructor happy
+                // otherwise we get things like 'protocol: matrix.org, pathname: 8448'
+                const url = new URL("https://" + serverName);
+                this._tryWellKnownDiscovery(url.hostname);
+            } catch (e) {
+                console.error("Problem parsing URL or unhandled error doing .well-known discovery");
```

Comment:
I think we've been converging on `console.log("message", e)` which does a sensible thing